### DISCUSSION TITLE
Help with adding an image texture to a ModeNode

### DISCUSSION THREAD

**Discussion by radcli14** [sceneview/sceneview#98] [Q&A]
Looking for help with my code below. I have a sphere, which I have as the model in the `ModelNode` named `node`, which I would like to have an image file super-imposed on top of. In this case, an image that represents the surface of the moon. The `setBaseColor` method at the top works, just to say the model loading and material seem to point to the right place. The image also seems to load properly. Setting the texture itself seems to have no effect. Any ideas?

```
// Here I am loading the material of my ModelNode and setting its color 
// using predefined floats for red, green, and blue
val material = node.modelInstance?.material?.filamentMaterialInstance
material?.setBaseColor(Float4(red, green, blue, 1.0f))

// Here I am loading a texture using an image file in my assets folder
val texture = TextureLoader.loadTexture(this, lifecycle,
    textureFileLocation = "textures/moonmap4k.jpg",
    type = TextureLoader.TextureType.NORMAL
)

// Here I want to set visual on the model node to be the image I just loaded
if (texture != null) {
    material?.setBaseTexture(texture)
}
```

**Comment by ThomasGorisse** [comment]
Hi,
You must set the base color index to 0.
Have a look at this sample:
https://github.com/SceneView/sceneform-android/blob/master/samples/image-texture/src/main/java/com/google/ar/sceneform/samples/imagetexture/MainActivity.java#L117

**Reply by radcli14** [reply]
Confirmed, that worked, just added a single line, see below. Thanks a ton!

```
if (texture != null) {
    node.modelInstance?.material?.setInt("baseColorIndex", 0);
    material?.setBaseTexture(texture)
}
```
